backend/src/main/java/gov/cdc/usds/simplereport/service/DataHubUploaderService.java

```diff
@@ -114,26 +205,92 @@ public String creatTestCVSForDataHub(String lastEndCreateOn) {
     // There is also the risk of the top action running multiple times concurrently.
     // ultimately, it would be nice if each row had an ID that could be dedupped on the server.
     @Transactional(readOnly = true)
-    public Map<String,String> uploadTestEventCVSToDataHub(final String apiKey, String lastEndCreateOn) {
+    public Map<String, String> uploadTestEventCSVToDataHub(final String apiKey, String lastEndCreateOn) {
         try {
-            this._createTestEventCSV(lastEndCreateOn);
-            this._uploadCSVDocument(apiKey);
+            this.createTestEventCSV(lastEndCreateOn);
+            this.uploadCSVDocument(apiKey);
 
-            return  Map.of(
+            return Map.of(
                     "result", "ok",
                     "lastTimestamp", this._nextTimestamp,
                     "rowsSent", String.valueOf(this._rowCount),
-                    "uploadResultId", this._uploadResultId,
+                    "uploadResultId", this._resultJson,
                     "message", this._warnMessage);
         } catch (RestClientException err) {
             return Map.of("result", "error",
-                    "message","Err: uploading csv to data-hub failed. error='" + err.toString() + "'");
+                    "message", "Err: uploading csv to data-hub failed. error='" + err.toString() + "'");
         } catch (IOException err) {
             return Map.of("result", "error",
                     "messsage", err.toString());
         } catch (NoResultException err) {
             return Map.of("result", "error",
-                    "message","No matching results for the given startupdateby param");
+                    "message", "No matching results for the given startupdateby param");
         }
     }
+
+    @Transactional
+    public void dataHubUploaderTask() {
+
+        // sanity check everything is configured correctly (dev likely will not be)
+        if (!_config.getUploadEnabled()) {
+            LOG.info("DataHubUploaderTask not running because simple-report.data-hub.uploadenabled is false");
+            return;
+        }
+
+        ArrayList<String> msgs = new ArrayList<>();
+        final int MIN_KEY_LENGTH = 24;
+        if (_config.getApiKey().length() < MIN_KEY_LENGTH) {
+            msgs.add("> DataHub API key is not configured.");
+        }
+        if (!_config.getUploadUrl().startsWith("https://")) {
+            msgs.add("> DataHub upload URL is not configured.");
+        }
+        if (!msgs.isEmpty()) {
+            sendSlackChannelMessage("DataHubUploader not run", msgs, true);
+            return;
+        }
+
+        // The start date is the last end date. Can be null for empty database.
+        Date lastTimestamp = getLatestRecordedTimestamp();
+        DataHubUpload newUpload = new DataHubUpload(_config)
```

Comment:
Responsibility shift: the constructor should set the state to INIT to begin with, and it should take in the timestamp as a constructor argument, so you can't modify that value after creating it.